### PR TITLE
KAFKA-10043:Some parameters will be overwritten which was configured …

### PR DESCRIPTION
**Pull request by SweetMojito**
…in consumer.config

*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by SweetMojito** [comment]
I dont kown how to link issues, so paste issues link here [https://issues.apache.org/jira/browse/KAFKA-10043](url)

**Comment by SweetMojito** [comment]
@ijuma Can you take a look at this (since you requested it in #8390  )?

**Comment by SweetMojito** [comment]
@guozhangwang Can you take a look at this?

**Comment by SweetMojito** [comment]
@ijuma Can you take a look at this (since you requested it in #8390  )?

**Comment by SweetMojito** [comment]
@rhauch  Can you take a look at this?

File: core/src/main/scala/kafka/tools/ConsumerPerformance.scala [hunk 1]
```diff
@@ -271,13 +271,22 @@ object ConsumerPerformance extends LazyLogging {
...
     val brokerHostsAndPorts = options.valueOf(if (options.has(bootstrapServerOpt)) bootstrapServerOpt else brokerListOpt)
     props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerHostsAndPorts)
-    props.put(ConsumerConfig.GROUP_ID_CONFIG, options.valueOf(groupIdOpt))
-    props.put(ConsumerConfig.RECEIVE_BUFFER_CONFIG, options.valueOf(socketBufferSizeOpt).toString)
-    props.put(ConsumerConfig.MAX_PARTITION_FETCH_BYTES_CONFIG, options.valueOf(fetchSizeOpt).toString)
-    props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, if (options.has(resetBeginningOffsetOpt)) "latest" else "earliest")
-    props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, classOf[ByteArrayDeserializer])
-    props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, classOf[ByteArrayDeserializer])
-    props.put(ConsumerConfig.CHECK_CRCS_CONFIG, "false")
+    if (props.getProperty(ConsumerConfig.GROUP_ID_CONFIG) == null)
```
Comment (chia7712) [inline core/src/main/scala/kafka/tools/ConsumerPerformance.scala:274]: Is ```props.putIfAbsent``` more better ?

**Review by chia7712 — COMMENTED** [review]
@itantiger Nice finding. I left a trivia comment for you. Also, could you add tests?

File: core/src/main/scala/kafka/tools/ConsumerPerformance.scala (hunk 1 above)
Comment (SweetMojito) [reply core/src/main/scala/kafka/tools/ConsumerPerformance.scala:274]: So sorry, no one replied to me before. I haven't paid attention to it for a long time

Comment (SweetMojito) [reply core/src/main/scala/kafka/tools/ConsumerPerformance.scala:274]: Hi @chia7712   This is my first submission, use `props.putIfAbsent` is better, but i dont what should do next？

Comment (chia7712) [reply core/src/main/scala/kafka/tools/ConsumerPerformance.scala:274]: > i dont what should do next？

Could you update the code (according to the reviews) and then push the changes to be a new commit. Also, please rebase your code (by trunk).

Comment (SweetMojito) [reply core/src/main/scala/kafka/tools/ConsumerPerformance.scala:274]: Do you mean i need to fork apache/kafka current trunk again? if that i will delete this fork.

Comment (SweetMojito) [reply core/src/main/scala/kafka/tools/ConsumerPerformance.scala:274]: I had forked once again and then updated the code (according to your suggestion). Can you take a look at this?
https://github.com/apache/kafka/pull/9790

**Pull request closed without merging**